src/main/java/htsjdk/samtools/DuplicateScoringStrategy.java

```diff
@@ -128,7 +128,7 @@ public static int compare(final SAMRecord rec1, final SAMRecord rec2, final Scor
         int cmp;
 
         // always prefer paired over non-paired
-        if (rec1.getReadPairedFlag() != rec2.getReadPairedFlag()) return rec1.getReadPairedFlag() ? 1 : -1;
+        if (rec1.getReadPairedFlag() != rec2.getReadPairedFlag()) return rec1.getReadPairedFlag() ? -1 : 1;
```

Comment:
yikes! 

that's a surprising bug. I guess we do not have many files that have paired and unpaired records together...

Can you add a test that covers this ?